x/emissions/keeper/msg_server_update_params_test.go

```diff
@@ -0,0 +1,56 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	keepertest "github.com/zeta-chain/zetacore/testutil/keeper"
+	"github.com/zeta-chain/zetacore/testutil/sample"
+	"github.com/zeta-chain/zetacore/x/emissions/keeper"
+	"github.com/zeta-chain/zetacore/x/emissions/types"
+)
+
+func TestMsgServer_UpdateParams(t *testing.T) {
+	t.Run("successfully update params", func(t *testing.T) {
+		k, ctx, _, _ := keepertest.EmissionsKeeper(t)
+		msgServer := keeper.NewMsgServerImpl(*k)
+
+		res, err := msgServer.UpdateParams(ctx, &types.MsgUpdateParams{
+			Authority: k.GetAuthority(),
+			Params:    types.DefaultParams(),
+		})
+
+		require.NoError(t, err)
+		require.Empty(t, res)
+		params, found := k.GetParams(ctx)
+		require.True(t, found)
+		require.Equal(t, types.DefaultParams(), params)
+	})
+
+	t.Run("fail for wrong authority", func(t *testing.T) {
+		k, ctx, _, _ := keepertest.EmissionsKeeper(t)
+		msgServer := keeper.NewMsgServerImpl(*k)
+
+		res, err := msgServer.UpdateParams(ctx, &types.MsgUpdateParams{
+			Authority: sample.AccAddress(),
+			Params:    types.DefaultParams(),
+		})
+
+		require.Error(t, err)
+		require.Nil(t, res)
+	})
+
+	t.Run("fail for invalid params", func(t *testing.T) {
+		k, ctx, _, _ := keepertest.EmissionsKeeper(t)
+		msgServer := keeper.NewMsgServerImpl(*k)
+		params := types.DefaultParams()
+		params.ValidatorEmissionPercentage = "-1.5"
+		res, err := msgServer.UpdateParams(ctx, &types.MsgUpdateParams{
+			Authority: k.GetAuthority(),
+			Params:    params,
+		})
+
+		require.Error(t, err)
+		require.Nil(t, res)
```

Comment:
NIT: I don't think we need to assert this value, it should be assume res would not be used in any case if an error is returned